services/src/main/java/org/keycloak/email/DefaultEmailSenderProvider.java

```diff
@@ -63,7 +63,11 @@ public DefaultEmailSenderProvider(KeycloakSession session) {
 
     @Override
     public void send(Map<String, String> config, UserModel user, String subject, String textBody, String htmlBody) throws EmailException {
-        send(config, retrieveEmailAddress(user), subject, textBody, htmlBody);
+        String address = retrieveEmailAddress(user);
+        if (address == null) {
+            throw new EmailException("No email address configured for the user", null);
```

Comment:
Looks like the second argument could be omitted here?

```suggestion
            throw new EmailException("No email address configured for the user");
```